### ISSUE TITLE
Receiving error uninitialized constant Chef::Provider::Sysvinit

### ISSUE THREAD

**Issue by gcimolson** [chef-cookbooks/mysql#370]
When running the install of mysql on an AWS instance with RHEL 7.1 I was receiving an error about an uninitialized constant.  I changed the lines in the z_provider_mapping.rb (client) from:

Chef::Platform.set platform: :amazon, resource: :mysql_service, provider: Chef::Provider::MysqlService::Sysvinit
to:

Chef::Platform.set platform: :amazon, resource: :mysql_service, provider: Chef::Provider::MysqlServiceSysvinit

After that the chef client completed without error although mysql doesn't appear to be starting correctly so I'll continue to debug that but just wanted to post this in case it helped someone else.

I'm new to this so sorry if this isn't the correct place to put this.

**Comment by someara** [comment]
The z_provider_mapping stuff is for Chef 11.x. It should be selecting the systemd provider on el7.
Can you gist a stack trace?
-s

**Comment by gcimolson** [comment]
My sincerest apologies.  When trying to get the recipe working on RHEL I had originally downgraded to an older version of the recipe and then when you put this newest one out I downloaded that one as well.  It appears when I did that I ended up with a dirty kitchen :-(  I still had some files hanging out there from the old version.  Thank you so much for your quick response and for supporting such a great cookbook.  I really appreciate it!  Sorry to have wasted any of your time.
